app/models/calculated_product_test.rb

```diff
@@ -129,4 +130,15 @@ def update_new_result_ids(res, qrda_id, new_rec_id)
     res_clone.save
   end
 
+  def read_qrda_file_contents(qrda_file)
+    qrda_file.respond_to?(:open) ? qrda_file.open.read : qrda_file.read
+  end
+
+  def should_generate_population?
+    !population_already_generated?
```

Comment:
I considered that approach (it actually used to all be inline!), but in lieu of comments explaining what we were doing and why, I went with the more verbose methods to explicitly explain to future me why and how we were bypassing automatic population generation.

If it's OK by you, I'd like to leave this.